docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,188 @@
+![vm-vm-standby](./Img/ssw-ha-test-plan-logic-standby.png)
+
+In case the traffic lands on active node, the path will be like below. 
+![vm-vm-active](./Img/ssw-ha-testplan-logic-active.png)
+
+
+## Test Methodology
+
+1. Test scenario is VM-to-VM traffic (we may add other critical scenarios, such as private link, in the future). 
+1. Open flow rules are configured on OVS Bridges to control the traffic flow:
+    * All packets sent out from Neighbor are only forwarded to DUT
+    * All packets sent out from DUT are forwarded to Neighbor and PTF
+    * All packets sent out from PTF are only forwarded to DUT
+1. Traffic will be sent from PTF docker, and sniffed on PTF docker by sonic-mgmt test scripts.
+1. Depending on the test cases, switchovers will be triggered and verified. 
+1. Control plane status, DPU counters, metering data, and flow table diff between active/standby will be measured for the tests.
+
+## Test Plan
+Assuming there is a pair of DPUs in the system, and at the step of test setup, we have DPU-1 as active, DPU-2 as standby. 
+
+Traffic will be injected by PTF through the T2 neighbors' OVS bridges. PTF will also sniff on the interfaces binding to the T2 neighbors.  
+
+For some of the cases, you will see two versions, one is traffic sending through DPU1, and the other one is traffic sending through DPU2. The purpose of  it is to verify:
+* Traffic shouldn’t be disrupted when failures happen on standby side.
+* Traffic should be tunneled if landing on standby side. 
+
+The name convention of a test case will be “\<Test Scenario\>-[Active|Standby]”, indicating the traffic is sent through the initial active or standby side. 
+
+### Module 1 Normal OP 
+| Case                      | Goal                                       | Test Steps                                                                                 | Expected Control Plane Behavior                               | Expected Data Plane Behavior                      |
+|---------------------------|--------------------------------------------|--------------------------------------------------------------------------------------------|--------------------------------------------------------------|---------------------------------------------------|
+| Normal OP – Active        | Verify normal operation in healthy state  | • Start I/O through Active side                                                 | DPU1 remains active, DPU2 remains standby.                   | T2 receives packets without disruption.         |
+| Normal OP – Standby       | Verify normal operation in healthy state  | • Start I/O through Standby side                                                | DPU1 remains active, DPU2 remains standby.                   | T2 receives packets without disruption from the active side.          |
```

Comment:
for baseline testing, we should also gathering latency data, and emit them via "test_reporting/telemetry" interfaces.

Reply:
Added.